### DISCUSSION TITLE
[V5] ENV_PATH not working

### DISCUSSION THREAD

**Discussion by GabrielGomesBarbosa** [adonisjs/core#3481] [Help]
Hi folks, i don't know whats happens but simply ENV_PATH stop working, im my project i have `.env` and `.env.local` files, and i use `.env.local` by the comand `ENV_PATH=.env.local node ace serve --watch` and its was working normaly, so i installed `cross-env` as developement dependency and my script on package.json was something like that:

```
  "scripts": {
    ...
    "dev": "cross-env ENV_PATH=.env.local node ace serve --watch",
    ....
  },
```

this works until today... but now a strange thing is happening, i will show

My env files:
![image](https://user-images.githubusercontent.com/37483241/148631308-73308246-3514-48b3-bda4-e0a406ebea38.png)

My Package.json:
![image](https://user-images.githubusercontent.com/37483241/148631345-11ec6cd6-b89e-4f12-8c3e-d2d6cad70aa5.png)

.env file content:
![image](https://user-images.githubusercontent.com/37483241/148631388-b5ad861c-efda-4450-99ad-ddff574f78c6.png)

.env.local file content:
![image](https://user-images.githubusercontent.com/37483241/148631427-6bbe0505-94ed-427b-a134-c70fb458637d.png)

Basically the only difference is the port, node environment and database variables, but on `.env` the port is `3334` and `.env.local` the port is `3333`, when i run the comand `ENV_PATH=.env.local node ace serve --watch`, is expected that the application use `.env.local` and run on port `3333`, but its not happens, look at the result:

![image](https://user-images.githubusercontent.com/37483241/148631537-7212da82-6914-477a-a8c5-bc9fca42782f.png)

Adonis get data from `.env` instead `.env.local` file, when i delete `.env` file, so the adonis get from `.env.local`, but if i delete `.env.local`  instead  `.env` i got error, that is the adonis load `.env.local` but get data from `.env`

PS: I just follow this documentation link https://docs.adonisjs.com/guides/environment-variables#do-not-commit-the-env-file

OS: Windows 10
AodnisJS: 5.3.4
Node: 14.18.1

**Comment by Julien-R44** [comment]
Are you sure that port 3333 is not already in use on your machine? Replace port 3333 by 3337 for example in your .env.local file and see if this port get assigned. If it is, it means that port 3333 is already used by a process on your machine

**Reply by GabrielGomesBarbosa** [reply]
Hi @Julien-R44 , thanks for your answer, i already tried it, but does not make sense, if i delete `.env` and execute `ENV_PATH=.env.local node ace serve --watch` it's run on port 3333 normally.

I did your test, i put port 3337, 3338 on my `.env.local` but it still get from `.env` when i execute `ENV_PATH=.env.local node ace serve --watch` command.

PS: I start a new project to test it, happen the same thing.
PS2: I test it on another machine and the same thing.

**Comment by Julien-R44** [comment]
I noticed the same behavior here, so I went and searched the code. In fact, the dev server will look for the three variables HOST, PORT, and TZ in the .env file, no matter the value of ENV_PATH : 
https://github.com/adonisjs/assembler/blob/cb877cf6ea975912b471ee4f39385aeb9493ccae/src/DevServer/index.ts#L98

So a quick way to fix this is to run the application like this: 
`ENV_PATH=./env.local PORT=3333 node ace serve --watch`

@thetutlage Maybe we could use `ENV_PATH` here instead of hardcoded ".env" path ? There might be a reason why this ".env" has been hardcoded, but I don't really see why at the moment
https://github.com/adonisjs/assembler/blob/cb877cf6ea975912b471ee4f39385aeb9493ccae/src/EnvParser/index.ts#L27

Or another solution could be to make the .env parsed by the application overwrite the rest? 
https://github.com/adonisjs/application/blob/develop/src/Application.ts#L381
